test/integration/controllers/users_controller_test.js

```diff
@@ -668,59 +687,82 @@ describe('Endpoint tests', () => {
   describe('user allergy tests', () => {
     let userToken = null;
 
-    before((done) => {
-      async.waterfall([
-        next => chai.request(usersRoutes)
-          .post('/users')
+    beforeEach((done) => {
+      chai.request(usersRoutes)
+        .post('/users')
+        .set('content-type', 'application/json')
+        .send({
+          email: 'user@email.com',
+          name: 'user',
+          password: '1234',
+        })
+        .end((err, registerRes) => {
+          should.not.exist(err);
+          registerRes.should.have.status(200);
+          should.exist(registerRes.body.token);
+          userToken = registerRes.body.token;
+          done();
+        });
+    });
+
+    afterEach((done) => {
+      usersModel.clean(done);
+    });
+
+    it('add an allergy successfully', (done) => {
+      async.auto({
+        addAllergies: callback => chai.request(usersRoutes)
+          .post('/users/allergies')
           .set('content-type', 'application/json')
-          .send({
-            email: 'user@email.com',
-            name: 'user',
-            password: '1234',
-          })
-          .end(next),
-        (res, next) => chai.request(usersRoutes)
-          .post('/users/login')
+          .set('token', userToken)
+          .send({ allergies: ['peanuts', 'beef'] })
+          .end(callback),
+        getUserInfo: ['addAllergies', (results, callback) => chai.request(usersRoutes)
+          .get('/users/user_info')
           .set('content-type', 'application/json')
-          .send({
-            email: 'user@email.com',
-            password: '1234',
-          })
-          .end(next),
-      ], (err, loginRes) => {
+          .set('token', userToken)
+          .end(callback),
+        ],
+      }, (err, { addAllergies, getUserInfo }) => {
         should.not.exist(err);
-        loginRes.should.have.status(200);
-        should.exist(loginRes.body.token);
-        userToken = loginRes.body.token;
+        addAllergies.should.have.status(200);
+        getUserInfo.should.have.status(200);
+        should.exist(getUserInfo.body.foodAllergies);
+        expect(getUserInfo.body.foodAllergies).to.include.members(['peanuts', 'beef']);
         done();
       });
     });
 
-    after((done) => {
-      usersModel.clean(done);
-    });
 
     it('add an allergy successfully', (done) => {
```

Comment:
do you mean, add same allergy again in test name

Reply:
your right, that's my bad, would you be okay if I fix this test name in the next commit?